<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.5.57">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Nick Kent">

<title>Content Summary #1 – Nick Kent</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../">
<script src="../../site_libs/quarto-html/quarto.js"></script>
<script src="../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>

  <script src="https://cdnjs.cloudflare.com/polyfill/v3/polyfill.min.js?features=es6"></script>
  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

<script type="text/javascript">
const typesetMath = (el) => {
  if (window.MathJax) {
    // MathJax Typeset
    window.MathJax.typeset([el]);
  } else if (window.katex) {
    // KaTeX Render
    var mathElements = el.getElementsByClassName("math");
    var macros = [];
    for (var i = 0; i < mathElements.length; i++) {
      var texText = mathElements[i].firstChild;
      if (mathElements[i].tagName == "SPAN") {
        window.katex.render(texText.data, mathElements[i], {
          displayMode: mathElements[i].classList.contains('display'),
          throwOnError: false,
          macros: macros,
          fleqn: false
        });
      }
    }
  }
}
window.Quarto = {
  typesetMath
};
</script>

<link rel="stylesheet" href="../../styles.css">
</head>

<body class="nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg " data-bs-theme="dark">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a class="navbar-brand" href="../../index.html">
    <span class="navbar-title">Nick Kent</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" role="menu" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link" href="../../index.html"> 
<span class="menu-text">Home</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../about.html"> 
<span class="menu-text">Resume</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../data_viz.html"> 
<span class="menu-text">Visualizing data</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../projects/Projects.html"> 
<span class="menu-text">Projects</span></a>
  </li>  
</ul>
          </div> <!-- /navcollapse -->
            <div class="quarto-navbar-tools">
    <a href="../../www.linkedin.com/in/nicholas-kent-300ab2277" title="" class="quarto-navigation-tool px-1" aria-label=""><i class="bi bi-linkedin"></i></a>
</div>
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#definitions" id="toc-definitions" class="nav-link active" data-scroll-target="#definitions">Definitions</a></li>
  <li><a href="#experiment" id="toc-experiment" class="nav-link" data-scroll-target="#experiment">Experiment</a>
  <ul class="collapse">
  <li><a href="#step-1-study-design" id="toc-step-1-study-design" class="nav-link" data-scroll-target="#step-1-study-design">Step 1: Study Design</a></li>
  <li><a href="#step-2-construct-genotype-for-n-observations" id="toc-step-2-construct-genotype-for-n-observations" class="nav-link" data-scroll-target="#step-2-construct-genotype-for-n-observations">Step 2: Construct genotype for n observations</a></li>
  <li><a href="#step-3-quality-control" id="toc-step-3-quality-control" class="nav-link" data-scroll-target="#step-3-quality-control">Step 3: Quality control</a></li>
  <li><a href="#step-4-multiple-testing-consideration" id="toc-step-4-multiple-testing-consideration" class="nav-link" data-scroll-target="#step-4-multiple-testing-consideration">Step 4: Multiple testing consideration</a></li>
  <li><a href="#step-5-create-quantitative-outcome-variable" id="toc-step-5-create-quantitative-outcome-variable" class="nav-link" data-scroll-target="#step-5-create-quantitative-outcome-variable">Step 5: Create quantitative outcome variable</a></li>
  <li><a href="#step-6-run-marginal-regression" id="toc-step-6-run-marginal-regression" class="nav-link" data-scroll-target="#step-6-run-marginal-regression">Step 6: Run marginal regression</a></li>
  <li><a href="#step-7-one-final-dataset" id="toc-step-7-one-final-dataset" class="nav-link" data-scroll-target="#step-7-one-final-dataset">Step 7: One final dataset</a></li>
  <li><a href="#step-8-create-visualizationsmanhattan-plotusing-ggplot-qq-plot" id="toc-step-8-create-visualizationsmanhattan-plotusing-ggplot-qq-plot" class="nav-link" data-scroll-target="#step-8-create-visualizationsmanhattan-plotusing-ggplot-qq-plot">Step 8: Create visualizations(Manhattan Plot(Using “ggplot”) + QQ Plot)</a></li>
  </ul></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Content Summary #1</h1>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Nick Kent </p>
          </div>
  </div>
    
  
    
  </div>
  


</header>


<section id="definitions" class="level1">
<h1>Definitions</h1>
<p><strong><em>Genotype</em></strong> - Genetic makeup of an organism, or an organisms specific DNA sequence.<br>
<br>
<strong><em>Phenotype</em></strong> - Observable characteristics of an organism resulting from it’s genotype.<br>
<br>
<strong><em>Nucleotide</em></strong> - The building blocks of DNA. 4 types: Adenine(A), Thymine(T), Guanine(G), Cystosine(C). Adenine and Thymine always pair together, and Guanine and Cystosine always pair together. Nucleotides play a large role in expressing genetic information. <strong>We tend to only look at oneside of the double-helix(ex: A G T C or T C A G) because we know that if an A is present on one side, the other side will be a T.</strong><br>
<br>
<strong><em>Gene</em></strong> - Specific sequence of nucleotides that code for a specific trait. Also can be thought of as a segment of DNA.<br>
<br>
<strong><em>Genetic Variant</em></strong> - Places where two humans genes differ. AT <strong>GC</strong> TA vs AT <strong>AT</strong> TA<br>
<br>
<strong><em>Single nucleotide variant(SNV)</em></strong> - A type of genetic variant that occurs when a single nucleotide (adenine, thymine, cytosine, or guanine) in the genome sequence differs.<br>
<br>
<strong><em>Allele</em></strong> - For a given SNV, an allele is the different possible nucleotides at that position(ex: A &amp; G or C &amp; T). At each position you have two alleles because you inherit one from each parent.<br>
<br>
<strong><em>Single Nucleotide Polymorphism(SNP)</em></strong> - Type of genetic variant where the minor allele has an MAF &gt; 1%. A SNP is a variation where a single nucleotide at a specific position in the DNA sequence differs between individuals.<br>
<br>
<strong><em>Major Allele</em></strong> - More common allele in a location/variant.<br>
<br>
<strong><em>Minor Allele</em></strong> - Less common allele in a location/variant. Minor allele is found by taking the combination of both alleles in each position(one from each parent) and then added up the total for each nucleotide in position one and then total for each nucleotide in position two, and then whichever one is least common is the minor allele.</p>
<table class="caption-top table">
<thead>
<tr class="header">
<th>Person</th>
<th>Father Allele</th>
<th>Mother Allele</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Person 1</td>
<td>A</td>
<td>A</td>
</tr>
<tr class="even">
<td>Person 2</td>
<td>A</td>
<td>G</td>
</tr>
<tr class="odd">
<td>Person 3</td>
<td>G</td>
<td>A</td>
</tr>
<tr class="even">
<td>Person 4</td>
<td>A</td>
<td>A</td>
</tr>
<tr class="odd">
<td>Person 5</td>
<td>G</td>
<td>G</td>
</tr>
<tr class="even">
<td>Person 6</td>
<td>A</td>
<td>A</td>
</tr>
</tbody>
</table>
<p><br>
Total A’s: 8<br>
Total G’s: 4<br>
Minor allele: G<br>
Major allele: A<br>
<br>
</p>
<table class="caption-top table">
<thead>
<tr class="header">
<th>Person</th>
<th>Minor Alleles</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Person 1</td>
<td>0</td>
</tr>
<tr class="even">
<td>Person 2</td>
<td>1</td>
</tr>
<tr class="odd">
<td>Person 3</td>
<td>1</td>
</tr>
<tr class="even">
<td>Person 4</td>
<td>0</td>
</tr>
<tr class="odd">
<td>Person 5</td>
<td>2</td>
</tr>
<tr class="even">
<td>Person 6</td>
<td>0</td>
</tr>
</tbody>
</table>
<p><br>
<br>
</p>
<p><strong><em>Minor Allele Frequency</em></strong> - Frequency of the minor allele in a population.</p>
</section>
<section id="experiment" class="level1">
<h1>Experiment</h1>
<p>We are going to do a fake genome wide association study to find out if there are any genetic variants that are correlated with someones IQ score. NOTE: all of the genetic data is going to be randomly generated, and every person is going to be assigned a random IQ score.</p>
<p><br>
<br>
</p>
<section id="step-1-study-design" class="level2">
<h2 class="anchored" data-anchor-id="step-1-study-design">Step 1: Study Design</h2>
<p>We are going to be using 100 “people” in this simulation study to walk through the desing of a GWAS. For each person we are going to gather 230,000 SNP’s(or nucleotides). This means that for each chromosome we will analyze 10,000 positions.<br>
<br>
</p>
</section>
<section id="step-2-construct-genotype-for-n-observations" class="level2">
<h2 class="anchored" data-anchor-id="step-2-construct-genotype-for-n-observations">Step 2: Construct genotype for n observations</h2>
<p>In each position(SNP) an observation either gets a AG, GA, AA, GG, each A/G represents what they got from each parent in that position(A or G from parent #1, and an A or G from parent #2).</p>
<p>First, for each SNP(or nucleotide) we have to come up with a probability of getting an A or G as one of your alleles. The probability of getting an A is a randomly chosen number from 0 to 1, and on the flip side the probability of getting a G is 1-(probability of getting an A).</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">42</span>)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Number of people, SNPs per chromosome, and chromosomes</span></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a>num_people <span class="ot">&lt;-</span> <span class="dv">100</span></span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a>num_snps_per_chr <span class="ot">&lt;-</span> <span class="dv">10000</span></span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a>num_chromosomes <span class="ot">&lt;-</span> <span class="dv">23</span></span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a>snp_grid <span class="ot">&lt;-</span> <span class="fu">expand_grid</span>(</span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">Chromosome =</span> <span class="dv">1</span><span class="sc">:</span>num_chromosomes,</span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a>  <span class="at">Position =</span> <span class="dv">1</span><span class="sc">:</span>num_snps_per_chr</span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span></span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a>    <span class="at">SNP_ID =</span> <span class="fu">row_number</span>(),</span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a>    <span class="at">prob_A =</span> <span class="fu">runif</span>(<span class="fu">n</span>(), <span class="dv">0</span>, <span class="dv">1</span>)</span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb1-16"><a href="#cb1-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">prob_G =</span> <span class="dv">1</span> <span class="sc">-</span> prob_A) </span>
<span id="cb1-17"><a href="#cb1-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-18"><a href="#cb1-18" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(snp_grid)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 6 × 5
  Chromosome Position SNP_ID prob_A prob_G
       &lt;int&gt;    &lt;int&gt;  &lt;int&gt;  &lt;dbl&gt;  &lt;dbl&gt;
1          1        1      1  0.915 0.0852
2          1        2      2  0.937 0.0629
3          1        3      3  0.286 0.714 
4          1        4      4  0.830 0.170 
5          1        5      5  0.642 0.358 
6          1        6      6  0.519 0.481 </code></pre>
</div>
</div>
<p>Now, using the probabilities for an A and G in each SNP position, we are going to assign everyone 2 alleles(one from each parent) at each position.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Create person data with SNPs and varying probabilities</span></span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>people_data <span class="ot">&lt;-</span> <span class="fu">tibble</span>(<span class="at">Person_ID =</span> <span class="dv">1</span><span class="sc">:</span>num_people) <span class="sc">%&gt;%</span></span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">crossing</span>(snp_grid) <span class="sc">%&gt;%</span></span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">Allele_1 =</span> <span class="fu">ifelse</span>(<span class="fu">runif</span>(<span class="fu">n</span>()) <span class="sc">&lt;</span> prob_A, <span class="st">"A"</span>, <span class="st">"G"</span>),</span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">Allele_2 =</span> <span class="fu">ifelse</span>(<span class="fu">runif</span>(<span class="fu">n</span>()) <span class="sc">&lt;</span> prob_A, <span class="st">"A"</span>, <span class="st">"G"</span>)</span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-9"><a href="#cb3-9" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(people_data)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 6 × 8
  Person_ID Chromosome Position SNP_ID prob_A prob_G Allele_1 Allele_2
      &lt;int&gt;      &lt;int&gt;    &lt;int&gt;  &lt;int&gt;  &lt;dbl&gt;  &lt;dbl&gt; &lt;chr&gt;    &lt;chr&gt;   
1         1          1        1      1  0.915 0.0852 A        A       
2         1          1        2      2  0.937 0.0629 A        G       
3         1          1        3      3  0.286 0.714  A        G       
4         1          1        4      4  0.830 0.170  G        A       
5         1          1        5      5  0.642 0.358  A        G       
6         1          1        6      6  0.519 0.481  A        G       </code></pre>
</div>
</div>
<p>Once everyone has two alleles at each position, we are going to count the number of each allele that everyone has at every position(ex. two A’s, one A, two G’s, etc.). Then we can add up across the whole population to figure out which allele is less common at that SNP, and we can also calculate the minor allele frequency at each SNP.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Identify the minor allele for each SNP</span></span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>minor_alleles <span class="ot">&lt;-</span> people_data <span class="sc">%&gt;%</span></span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(SNP_ID) <span class="sc">%&gt;%</span></span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">count_A =</span> <span class="fu">sum</span>(Allele_1 <span class="sc">==</span> <span class="st">"A"</span> <span class="sc">|</span> Allele_2 <span class="sc">==</span> <span class="st">"A"</span>), </span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">count_G =</span> <span class="fu">sum</span>(Allele_1 <span class="sc">==</span> <span class="st">"G"</span> <span class="sc">|</span> Allele_2 <span class="sc">==</span> <span class="st">"G"</span>)</span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb5-9"><a href="#cb5-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">Minor_Allele =</span> <span class="fu">if_else</span>(count_A <span class="sc">&lt;</span> count_G, <span class="st">"A"</span>, <span class="st">"G"</span>), </span>
<span id="cb5-10"><a href="#cb5-10" aria-hidden="true" tabindex="-1"></a>    <span class="at">MAF =</span> <span class="fu">pmin</span>(count_A, count_G) <span class="sc">/</span> (<span class="dv">2</span> <span class="sc">*</span> num_people)</span>
<span id="cb5-11"><a href="#cb5-11" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb5-12"><a href="#cb5-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(SNP_ID, Minor_Allele, MAF) <span class="sc">%&gt;%</span></span>
<span id="cb5-13"><a href="#cb5-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>()</span>
<span id="cb5-14"><a href="#cb5-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-15"><a href="#cb5-15" aria-hidden="true" tabindex="-1"></a><span class="co"># </span><span class="al">NOTE</span><span class="co">: This is part of quality control</span></span>
<span id="cb5-16"><a href="#cb5-16" aria-hidden="true" tabindex="-1"></a>minor_alleles <span class="ot">&lt;-</span> minor_alleles <span class="sc">%&gt;%</span> </span>
<span id="cb5-17"><a href="#cb5-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(MAF <span class="sc">!=</span> <span class="dv">0</span>)</span>
<span id="cb5-18"><a href="#cb5-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-19"><a href="#cb5-19" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(minor_alleles)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 6 × 3
# Groups:   SNP_ID [6]
  SNP_ID Minor_Allele   MAF
   &lt;int&gt; &lt;chr&gt;        &lt;dbl&gt;
1      1 G            0.075
2      2 G            0.09 
3      3 A            0.235
4      4 G            0.19 
5      5 G            0.3  
6      6 A            0.35 </code></pre>
</div>
</div>
<p>Once we know the minor allele for each SNP, we can calculate the number of minor alleles that each person has at each position. These will become our predictor variables for our regression models.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Counting the minor alleles at each SNP position</span></span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a>minor_allele_counts <span class="ot">&lt;-</span> people_data <span class="sc">%&gt;%</span></span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(minor_alleles, <span class="at">by =</span> <span class="st">"SNP_ID"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">Minor_Allele_Count =</span> </span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a>      (Minor_Allele <span class="sc">==</span> Allele_1) <span class="sc">+</span> (Minor_Allele <span class="sc">==</span> Allele_2)  <span class="co"># Vectorized count</span></span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb7-8"><a href="#cb7-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(Person_ID, SNP_ID, Minor_Allele_Count) <span class="sc">%&gt;%</span>  <span class="co"># Keep only necessary columns</span></span>
<span id="cb7-9"><a href="#cb7-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> SNP_ID, <span class="at">values_from =</span> Minor_Allele_Count)</span>
<span id="cb7-10"><a href="#cb7-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-11"><a href="#cb7-11" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(minor_allele_counts)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 6 × 230,001
  Person_ID   `1`   `2`   `3`   `4`   `5`   `6`   `7`   `8`   `9`  `10`  `11`
      &lt;int&gt; &lt;int&gt; &lt;int&gt; &lt;int&gt; &lt;int&gt; &lt;int&gt; &lt;int&gt; &lt;int&gt; &lt;int&gt; &lt;int&gt; &lt;int&gt; &lt;int&gt;
1         1     0     1     1     1     1     1     0     1     2     1     2
2         2     0     0     1     2     1     2     1     1     1     1     1
3         3     0     0     0     1     1     0     1     0     1     1     2
4         4     0     0     0     1     0     1     1     1     1     0     0
5         5     0     0     0     0     1     0     2     0     1     1     1
6         6     0     0     0     0     1     2     1     0     1     1     1
# ℹ 229,989 more variables: `12` &lt;int&gt;, `13` &lt;int&gt;, `14` &lt;int&gt;, `15` &lt;int&gt;,
#   `16` &lt;int&gt;, `17` &lt;int&gt;, `18` &lt;int&gt;, `19` &lt;int&gt;, `20` &lt;int&gt;, `21` &lt;int&gt;,
#   `22` &lt;int&gt;, `23` &lt;int&gt;, `24` &lt;int&gt;, `25` &lt;int&gt;, `26` &lt;int&gt;, `27` &lt;int&gt;,
#   `28` &lt;int&gt;, `29` &lt;int&gt;, `30` &lt;int&gt;, `31` &lt;int&gt;, `32` &lt;int&gt;, `33` &lt;int&gt;,
#   `34` &lt;int&gt;, `35` &lt;int&gt;, `36` &lt;int&gt;, `37` &lt;int&gt;, `38` &lt;int&gt;, `39` &lt;int&gt;,
#   `40` &lt;int&gt;, `41` &lt;int&gt;, `42` &lt;int&gt;, `43` &lt;int&gt;, `44` &lt;int&gt;, `45` &lt;int&gt;,
#   `46` &lt;int&gt;, `47` &lt;int&gt;, `48` &lt;int&gt;, `49` &lt;int&gt;, `50` &lt;int&gt;, `51` &lt;int&gt;, …</code></pre>
</div>
</div>
<p><br>
<br>
</p>
</section>
<section id="step-3-quality-control" class="level2">
<h2 class="anchored" data-anchor-id="step-3-quality-control">Step 3: Quality control</h2>
<p>We want to filter out all of the SNP that dont have a minor allele, meaning everyone has the same alleles at that position.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a>filtered_snps <span class="ot">&lt;-</span> minor_alleles<span class="sc">$</span>SNP_ID  <span class="co"># SNPs with nonzero MAF</span></span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove columns that are no longer in minor_allele_counts due to filtering</span></span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a>valid_columns <span class="ot">&lt;-</span> <span class="fu">intersect</span>(<span class="fu">names</span>(minor_allele_counts), <span class="fu">as.character</span>(filtered_snps))</span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-6"><a href="#cb9-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Select only the valid columns (Person_ID + SNP columns) and remove columns that are all NA</span></span>
<span id="cb9-7"><a href="#cb9-7" aria-hidden="true" tabindex="-1"></a>minor_allele_counts <span class="ot">&lt;-</span> minor_allele_counts <span class="sc">%&gt;%</span></span>
<span id="cb9-8"><a href="#cb9-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(Person_ID, <span class="fu">all_of</span>(valid_columns)) <span class="sc">%&gt;%</span></span>
<span id="cb9-9"><a href="#cb9-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="fu">where</span>(<span class="sc">~</span> <span class="sc">!</span><span class="fu">all</span>(<span class="fu">is.na</span>(.))))</span>
<span id="cb9-10"><a href="#cb9-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-11"><a href="#cb9-11" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(minor_allele_counts)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 6 × 227,591
  Person_ID   `1`   `2`   `3`   `4`   `5`   `6`   `7`   `8`   `9`  `10`  `11`
      &lt;int&gt; &lt;int&gt; &lt;int&gt; &lt;int&gt; &lt;int&gt; &lt;int&gt; &lt;int&gt; &lt;int&gt; &lt;int&gt; &lt;int&gt; &lt;int&gt; &lt;int&gt;
1         1     0     1     1     1     1     1     0     1     2     1     2
2         2     0     0     1     2     1     2     1     1     1     1     1
3         3     0     0     0     1     1     0     1     0     1     1     2
4         4     0     0     0     1     0     1     1     1     1     0     0
5         5     0     0     0     0     1     0     2     0     1     1     1
6         6     0     0     0     0     1     2     1     0     1     1     1
# ℹ 227,579 more variables: `12` &lt;int&gt;, `13` &lt;int&gt;, `14` &lt;int&gt;, `15` &lt;int&gt;,
#   `16` &lt;int&gt;, `17` &lt;int&gt;, `18` &lt;int&gt;, `19` &lt;int&gt;, `20` &lt;int&gt;, `21` &lt;int&gt;,
#   `22` &lt;int&gt;, `23` &lt;int&gt;, `24` &lt;int&gt;, `25` &lt;int&gt;, `26` &lt;int&gt;, `27` &lt;int&gt;,
#   `28` &lt;int&gt;, `29` &lt;int&gt;, `30` &lt;int&gt;, `31` &lt;int&gt;, `32` &lt;int&gt;, `33` &lt;int&gt;,
#   `34` &lt;int&gt;, `35` &lt;int&gt;, `36` &lt;int&gt;, `37` &lt;int&gt;, `38` &lt;int&gt;, `39` &lt;int&gt;,
#   `40` &lt;int&gt;, `41` &lt;int&gt;, `42` &lt;int&gt;, `43` &lt;int&gt;, `44` &lt;int&gt;, `45` &lt;int&gt;,
#   `46` &lt;int&gt;, `47` &lt;int&gt;, `48` &lt;int&gt;, `49` &lt;int&gt;, `50` &lt;int&gt;, `51` &lt;int&gt;, …</code></pre>
</div>
</div>
<p><br>
<br>
</p>
</section>
<section id="step-4-multiple-testing-consideration" class="level2">
<h2 class="anchored" data-anchor-id="step-4-multiple-testing-consideration">Step 4: Multiple testing consideration</h2>
<p>We need to come up with a new significance threshold because we are running so many regression models we need to avoid getting too many false positives. For example, if we had a significance threshold of <span class="math inline">\(\alpha\)</span> = 0.05(which is the norm in many fields) the probability of getting at least one type one error(or false positive): P(At least one T1E) = 1 - P(no T1E’s) = <span class="math inline">\(1 - (1- \alpha ) ^{230000}\)</span> <span class="math inline">\(\approx\)</span> 1. This means that if we were to use a significance threshold of <span class="math inline">\(\alpha\)</span> = 0.05 for 230,000 tests, we are almost guaranteed to get at least one false positive.<br>
<br>
Bonferroni solution: One very widely used approach for adjusting for multiple testing is known as the Bonferroni Correction. This is one way to help keep our the probability of at least one type 1 error(AKA. Family wise error rate) low or as low as we want it. We calculate our new significance threshold by dividing our desired family-wise error rate by the number of hypothesis tests that we will conduct.<br>
<br>
<br>
Significance threshold (<span class="math inline">\(\alpha\)</span>)= 0.05 / 227642 = <span class="math inline">\(2.196431e^{-07}\)</span><br>
<br>
P(at least one T1E) = <span class="math inline">\(1 - (1 - 2.196431e^{-07})^{230000}\)</span> <span class="math inline">\(\approx\)</span> 0.05<br>
<br>
So with this significance threshold we can keep our family wise error rate close to 0.05.<br>
<br>
</p>
</section>
<section id="step-5-create-quantitative-outcome-variable" class="level2">
<h2 class="anchored" data-anchor-id="step-5-create-quantitative-outcome-variable">Step 5: Create quantitative outcome variable</h2>
<p>Now, we are going to create a random outcome variable. In this case, everyone is going to receive a fake IQ score that is randomly chosen from a distribution centered at 100.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a>minor_allele_counts <span class="ot">&lt;-</span> minor_allele_counts <span class="sc">%&gt;%</span> </span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">IQ =</span> <span class="fu">rnorm</span>(<span class="fu">n</span>(), <span class="at">mean =</span> <span class="dv">100</span>, <span class="at">sd =</span> <span class="dv">15</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p><br>
So, the hypothesis we are going to be testing at each SNP is:<br>
<span class="math inline">\(H_O\)</span> = There IS NO association between SNP of interest and outcome variable, IQ score.<br>
<span class="math inline">\(H_A\)</span> = There IS association between SNP of interest and outcome variable, IQ score.<br>
Using our calculated significance threshold of <span class="math inline">\(2.196431e^{-07}\)</span>, we will either accept/reject the null hypothesis for each SNP.<br>
<br>
</p>
</section>
<section id="step-6-run-marginal-regression" class="level2">
<h2 class="anchored" data-anchor-id="step-6-run-marginal-regression">Step 6: Run marginal regression</h2>
<p>Now, using the count for minor alleles at each SNP, we are going to create 230,000 regression models that estimate IQ score from minor allele counts at each SNP in order to test the hypothesis, this will allow us to find out if there is association between any SNP’s and IQ score. The regression formula for each SNP will look something like:<br>
<br>
<span class="math inline">\(E[IQ | \text{Minor Allele Count}] = \beta_0 + \beta_1 \cdot \text{(Number of Minor Alleles at Certain SNP)}\)</span></p>
<div class="cell">
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a>IQ <span class="ot">&lt;-</span> minor_allele_counts<span class="sc">$</span>IQ</span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a>X <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(minor_allele_counts <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="sc">-</span>IQ, <span class="sc">-</span>Person_ID))  <span class="co"># Convert predictors to a matrix</span></span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Initialize vectors to store results</span></span>
<span id="cb12-5"><a href="#cb12-5" aria-hidden="true" tabindex="-1"></a>num_snps <span class="ot">&lt;-</span> <span class="fu">ncol</span>(X)</span>
<span id="cb12-6"><a href="#cb12-6" aria-hidden="true" tabindex="-1"></a>all_betas <span class="ot">&lt;-</span> <span class="fu">numeric</span>(num_snps)</span>
<span id="cb12-7"><a href="#cb12-7" aria-hidden="true" tabindex="-1"></a>all_ses <span class="ot">&lt;-</span> <span class="fu">numeric</span>(num_snps)</span>
<span id="cb12-8"><a href="#cb12-8" aria-hidden="true" tabindex="-1"></a>all_tstats <span class="ot">&lt;-</span> <span class="fu">numeric</span>(num_snps)</span>
<span id="cb12-9"><a href="#cb12-9" aria-hidden="true" tabindex="-1"></a>all_pvals <span class="ot">&lt;-</span> <span class="fu">numeric</span>(num_snps)</span>
<span id="cb12-10"><a href="#cb12-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-11"><a href="#cb12-11" aria-hidden="true" tabindex="-1"></a><span class="co"># Run marginal regressions efficiently</span></span>
<span id="cb12-12"><a href="#cb12-12" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>num_snps) {</span>
<span id="cb12-13"><a href="#cb12-13" aria-hidden="true" tabindex="-1"></a>  model <span class="ot">&lt;-</span> <span class="fu">lm</span>(IQ <span class="sc">~</span> X[, i])  <span class="co"># Fit a simple regression model for each predictor</span></span>
<span id="cb12-14"><a href="#cb12-14" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb12-15"><a href="#cb12-15" aria-hidden="true" tabindex="-1"></a>  coefinfo <span class="ot">&lt;-</span> <span class="fu">tidy</span>(model)  <span class="co"># Extract coefficients</span></span>
<span id="cb12-16"><a href="#cb12-16" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb12-17"><a href="#cb12-17" aria-hidden="true" tabindex="-1"></a>  all_betas[i] <span class="ot">&lt;-</span> coefinfo<span class="sc">$</span>estimate[<span class="dv">2</span>]</span>
<span id="cb12-18"><a href="#cb12-18" aria-hidden="true" tabindex="-1"></a>  all_ses[i] <span class="ot">&lt;-</span> coefinfo<span class="sc">$</span>std.error[<span class="dv">2</span>]</span>
<span id="cb12-19"><a href="#cb12-19" aria-hidden="true" tabindex="-1"></a>  all_tstats[i] <span class="ot">&lt;-</span> coefinfo<span class="sc">$</span>statistic[<span class="dv">2</span>]</span>
<span id="cb12-20"><a href="#cb12-20" aria-hidden="true" tabindex="-1"></a>  all_pvals[i] <span class="ot">&lt;-</span> coefinfo<span class="sc">$</span>p.value[<span class="dv">2</span>]</span>
<span id="cb12-21"><a href="#cb12-21" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb12-22"><a href="#cb12-22" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (i <span class="sc">%%</span> <span class="dv">10000</span> <span class="sc">==</span> <span class="dv">0</span>) <span class="fu">print</span>(<span class="fu">paste</span>(<span class="st">"Analyzing predictor"</span>, i))  <span class="co"># Progress update</span></span>
<span id="cb12-23"><a href="#cb12-23" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "Analyzing predictor 10000"
[1] "Analyzing predictor 20000"
[1] "Analyzing predictor 30000"
[1] "Analyzing predictor 40000"
[1] "Analyzing predictor 50000"
[1] "Analyzing predictor 60000"
[1] "Analyzing predictor 70000"
[1] "Analyzing predictor 80000"
[1] "Analyzing predictor 90000"
[1] "Analyzing predictor 100000"
[1] "Analyzing predictor 110000"
[1] "Analyzing predictor 120000"
[1] "Analyzing predictor 130000"
[1] "Analyzing predictor 140000"
[1] "Analyzing predictor 150000"
[1] "Analyzing predictor 160000"
[1] "Analyzing predictor 170000"
[1] "Analyzing predictor 180000"
[1] "Analyzing predictor 190000"
[1] "Analyzing predictor 200000"
[1] "Analyzing predictor 210000"
[1] "Analyzing predictor 220000"</code></pre>
</div>
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Create results data frame</span></span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a>results <span class="ot">&lt;-</span> <span class="fu">tibble</span>(</span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">SNP_ID =</span> <span class="fu">as.integer</span>(<span class="fu">colnames</span>(X)),</span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">Beta =</span> all_betas,</span>
<span id="cb14-5"><a href="#cb14-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">SE =</span> all_ses,</span>
<span id="cb14-6"><a href="#cb14-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">T_Stat =</span> all_tstats,</span>
<span id="cb14-7"><a href="#cb14-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">P_Value =</span> all_pvals</span>
<span id="cb14-8"><a href="#cb14-8" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb14-9"><a href="#cb14-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-10"><a href="#cb14-10" aria-hidden="true" tabindex="-1"></a><span class="co"># View top predictors by significance</span></span>
<span id="cb14-11"><a href="#cb14-11" aria-hidden="true" tabindex="-1"></a>results <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(P_Value) <span class="sc">%&gt;%</span> <span class="fu">head</span>(<span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 10 × 5
   SNP_ID   Beta    SE T_Stat    P_Value
    &lt;int&gt;  &lt;dbl&gt; &lt;dbl&gt;  &lt;dbl&gt;      &lt;dbl&gt;
 1  39087  -9.05  1.92  -4.72 0.00000794
 2 120849  12.9   2.73   4.71 0.00000804
 3  37580  10.4   2.23   4.69 0.00000897
 4  38818   8.51  1.83   4.65 0.0000103 
 5  82446  12.5   2.73   4.59 0.0000132 
 6  78227   8.59  1.89   4.53 0.0000164 
 7 208474  -9.16  2.05  -4.47 0.0000214 
 8  91032 -10.3   2.30  -4.46 0.0000219 
 9  84519  18.6   4.19   4.44 0.0000234 
10  70805  10.7   2.42   4.44 0.0000240 </code></pre>
</div>
</div>
<p>Above are the ten most significant results from our study.<br>
<br>
</p>
</section>
<section id="step-7-one-final-dataset" class="level2">
<h2 class="anchored" data-anchor-id="step-7-one-final-dataset">Step 7: One final dataset</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a>final_results <span class="ot">&lt;-</span> minor_alleles <span class="sc">%&gt;%</span> </span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(results, <span class="at">by=</span><span class="st">"SNP_ID"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(snp_grid, <span class="at">by=</span><span class="st">"SNP_ID"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(SNP_ID, Minor_Allele, MAF, Chromosome, Position, P_Value)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p><br>
<br>
</p>
</section>
<section id="step-8-create-visualizationsmanhattan-plotusing-ggplot-qq-plot" class="level2">
<h2 class="anchored" data-anchor-id="step-8-create-visualizationsmanhattan-plotusing-ggplot-qq-plot">Step 8: Create visualizations(Manhattan Plot(Using “ggplot”) + QQ Plot)</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a>final_results <span class="sc">%&gt;%</span> </span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Chromosome =</span> <span class="fu">as.factor</span>(<span class="st">`</span><span class="at">Chromosome</span><span class="st">`</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> SNP_ID, <span class="at">y =</span> <span class="sc">-</span><span class="fu">log10</span>(P_Value), <span class="at">color =</span> Chromosome))<span class="sc">+</span></span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>()<span class="sc">+</span></span>
<span id="cb17-5"><a href="#cb17-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="sc">-</span><span class="fu">log10</span>(<span class="fl">2.196431e-07</span>), <span class="at">linetype =</span> <span class="st">"dashed"</span>, <span class="at">color =</span> <span class="st">"Pink"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="ContentSummary_NoAssoc_files/figure-html/manhattan-plot-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="fu">qq</span>(final_results<span class="sc">$</span>P_Value, <span class="at">main =</span> <span class="st">"Q-Q plot of GWAS p-values"</span>, <span class="at">xlim =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">7</span>), <span class="at">ylim =</span> <span class="fu">c</span>(<span class="dv">0</span>,</span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a>    <span class="dv">12</span>), <span class="at">pch =</span> <span class="dv">18</span>, <span class="at">col =</span> <span class="st">"blue4"</span>, <span class="at">cex =</span> <span class="dv">1</span>, <span class="at">las =</span> <span class="dv">1</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="ContentSummary_NoAssoc_files/figure-html/qq-plot-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>


</section>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    // For code content inside modals, clipBoardJS needs to be initialized with a container option
    // TODO: Check when it could be a function (https://github.com/zenorocha/clipboard.js/issues/860)
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->




</body></html>